src/main/app/form/validation.ts

```diff
@@ -59,28 +59,40 @@ export const doesArrayHaveValues: Validator = value => {
 };
 
 export const isDateInputNotFilled: DateValidator = date => {
-  const invalid = 'invalidDate';
-  if (!date) {
-    return invalid;
+  if (!date || (isEmpty(date.day) && isEmpty(date.month) && isEmpty(date.year))) {
+    return 'incompleteDayAndMonthAndYear';
   }
 
-  for (const value in date) {
-    if (isNaN(+date[value])) {
-      return invalid;
-    }
+  if (isEmpty(date.day) && isEmpty(date.month)) {
+    return 'incompleteDayAndMonth';
   }
 
-  if (isEmpty(date.day || date.month || date.year)) {
-    return invalid;
+  if (isEmpty(date.day) && isEmpty(date.year)) {
+    return 'incompleteDayAndYear';
+  }
+
+  if (isEmpty(date.month) && isEmpty(date.year)) {
+    return 'incompleteMonthAndYear';
+  }
+
+  if (isEmpty(date.day)) {
+    return 'incompleteDay';
+  }
+  if (isEmpty(date.month)) {
+    return 'incompleteMonth';
+  }
+  if (isEmpty(date.year)) {
+    return 'incompleteYear';
```

Comment:
I think this block would be better if you used else ifs instead of multiple ifs.

Reply:
Fixed now